macros/src/main/scala/CostMetric.scala

```diff
@@ -113,18 +113,16 @@ object ExternalMetric extends CostMetricCompanion {
 /** The current default metric in barstools, re-defined by Donggyu. */
 // TODO: write tests for this function to make sure it selects the right things
 object DefaultMetric extends CostMetric with CostMetricCompanion {
-  override def cost(mem: Macro, lib: Macro): Option[BigInt] = {
+  override def cost(mem: Macro, lib: Macro): Option[Double] = {
     val memMask = mem.src.ports map (_.maskGran) find (_.isDefined) map (_.get)
     val libMask = lib.src.ports map (_.maskGran) find (_.isDefined) map (_.get)
     val memWidth = (memMask, libMask) match {
-      case (Some(1), Some(1)) | (None, _) => mem.src.width
+      case (Some(_), Some(1)) | (None, _) => mem.src.width
```

Comment:
Was this just a bug? Not sure I understand the change.

Reply:
If the library has a mask granularity of 1 we can fit any source/design memory in it so we can just consider its width the nominal width (rather than some function of its mask width and the library mask width).
This function is here for some edge that someone was trying to work around but in practice never shows up. This code is slightly more correct but the second case I think is still not correct.